Compilation/Linking error when using quite minimal Godot build

### Godot version

4.0.beta.custom_build. 28a4eec9a

### System information

Ubuntu 22.04 - Nvidia GTX 970, Gnome shell 42 X11

### Issue description

```
scons -j7  dev_build=yes use_asan=yes use_ubsan=yes CCFLAGS="-fsanitize=shift,shift-exponent,integer-divide-by-zero,unreachable,vla-bound,null,return,signed-integer-overflow,bounds,float-divide-by-zero,float-cast-overflow,nonnull-attribute,returns-nonnull-attribute,bool,enum,vptr" use_llvm=yes module_basis_universal_enabled=no module_bmp_enabled=no module_camera_enabled=no module_csg_enabled=no module_cvtt_enabled=no module_dds_enabled=no module_denoise_enabled=no module_enet_enabled=no module_etcpak_enabled=no module_glslang_enabled=no module_gltf_enabled=no module_gridmap_enabled=no module_hdr_enabled=no module_jpg_enabled=no module_jsonrpc_enabled=no module_lightmapper_rd_enabled=no module_mbedtls_enabled=no module_meshoptimizer_enabled=no module_minimp3_enabled=no module_mobile_vr_enabled=no module_msdfgen_enabled=no module_multiplayer_enabled=no module_noise_enabled=no module_ogg_enabled=no module_openxr_enabled=no module_raycast_enabled=no module_regex_enabled=no module_squish_enabled=no graphite=no module_tga_enabled=no module_theora_enabled=no module_tinyexr_enabled=no module_upnp_enabled=no module_vhacd_enabled=no module_vorbis_enabled=no module_webp_enabled=no module_webrtc_enabled=no module_websocket_enabled=no module_webxr_enabled=no module_xatlas_unwrap_enabled=no

```
produce this errors
```
/usr/bin/ld: /usr/bin/ld: DWARF error: invalid or unhandled FORM value: 0x23
modules/text_server_adv/libharfbuzz_builtin.linuxbsd.editor.dev.x86_64.llvm.san.a(hb-face.linuxbsd.editor.dev.x86_64.llvm.san.o): in function `hb_shaper_lazy_loader_t<hb_face_t, 1u, hb_graphite2_face_data_t>::destroy(hb_graphite2_face_data_t*)':
hb-face.cc:(.text._ZN23hb_shaper_lazy_loader_tI9hb_face_tLj1E24hb_graphite2_face_data_tE7destroyEPS1_[_ZN23hb_shaper_lazy_loader_tI9hb_face_tLj1E24hb_graphite2_face_data_tE7destroyEPS1_]+0x19): undefined reference to `_hb_graphite2_shaper_face_data_destroy'
/usr/bin/ld: /usr/bin/ld: DWARF error: invalid or unhandled FORM value: 0x23
modules/text_server_adv/libharfbuzz_builtin.linuxbsd.editor.dev.x86_64.llvm.san.a(hb-font.linuxbsd.editor.dev.x86_64.llvm.san.o): in function `hb_shaper_lazy_loader_t<hb_font_t, 1u, hb_graphite2_font_data_t>::destroy(hb_graphite2_font_data_t*)':
hb-font.cc:(.text._ZN23hb_shaper_lazy_loader_tI9hb_font_tLj1E24hb_graphite2_font_data_tE7destroyEPS1_[_ZN23hb_shaper_lazy_loader_tI9hb_font_tLj1E24hb_graphite2_font_data_tE7destroyEPS1_]+0x19): undefined reference to `_hb_graphite2_shaper_font_data_destroy'
/usr/bin/ld: /usr/bin/ld: DWARF error: invalid or unhandled FORM value: 0x23
modules/text_server_adv/libharfbuzz_builtin.linuxbsd.editor.dev.x86_64.llvm.san.a(hb-shape-plan.linuxbsd.editor.dev.x86_64.llvm.san.o): in function `hb_shape_plan_key_t::init(bool, hb_face_t*, hb_segment_properties_t const*, hb_feature_t const*, unsigned int, int const*, unsigned int, char const* const*)':
hb-shape-plan.cc:(.text+0x9d3): undefined reference to `_hb_graphite2_shape'
/usr/bin/ld: hb-shape-plan.cc:(.text+0x10c6): undefined reference to `_hb_graphite2_shape'
/usr/bin/ld: hb-shape-plan.cc:(.text+0x1216): undefined reference to `_hb_graphite2_shape'
/usr/bin/ld: modules/text_server_adv/libharfbuzz_builtin.linuxbsd.editor.dev.x86_64.llvm.san.a(hb-shape-plan.linuxbsd.editor.dev.x86_64.llvm.san.o): in function `_hb_shape_plan_execute_internal(hb_shape_plan_t*, hb_font_t*, hb_buffer_t*, hb_feature_t const*, unsigned int)':
hb-shape-plan.cc:(.text+0x47bc): undefined reference to `_hb_graphite2_shape'
/usr/bin/ld: hb-shape-plan.cc:(.text+0x48f5): undefined reference to `_hb_graphite2_shape'
/usr/bin/ld: modules/text_server_adv/libharfbuzz_builtin.linuxbsd.editor.dev.x86_64.llvm.san.a(hb-shape-plan.linuxbsd.editor.dev.x86_64.llvm.san.o): in function `hb_shaper_lazy_loader_t<hb_font_t, 1u, hb_graphite2_font_data_t>::create(hb_font_t*)':
hb-shape-plan.cc:(.text._ZN23hb_shaper_lazy_loader_tI9hb_font_tLj1E24hb_graphite2_font_data_tE6createEPS0_[_ZN23hb_shaper_lazy_loader_tI9hb_font_tLj1E24hb_graphite2_font_data_tE6createEPS0_]+0x19): undefined reference to `_hb_graphite2_shaper_font_data_create'
/usr/bin/ld: modules/text_server_adv/libharfbuzz_builtin.linuxbsd.editor.dev.x86_64.llvm.san.a(hb-shape-plan.linuxbsd.editor.dev.x86_64.llvm.san.o): in function `hb_shaper_lazy_loader_t<hb_face_t, 1u, hb_graphite2_face_data_t>::create(hb_face_t*)':
hb-shape-plan.cc:(.text._ZN23hb_shaper_lazy_loader_tI9hb_face_tLj1E24hb_graphite2_face_data_tE6createEPS0_[_ZN23hb_shaper_lazy_loader_tI9hb_face_tLj1E24hb_graphite2_face_data_tE6createEPS0_]+0x19): undefined reference to `_hb_graphite2_shaper_face_data_create'
/usr/bin/ld: /usr/bin/ld: DWARF error: invalid or unhandled FORM value: 0x23
modules/text_server_adv/libharfbuzz_builtin.linuxbsd.editor.dev.x86_64.llvm.san.a(hb-shaper.linuxbsd.editor.dev.x86_64.llvm.san.o):(.data.rel.ro+0x70): undefined reference to `_hb_graphite2_shape'
/usr/bin/ld: bin/godot.linuxbsd.editor.dev.x86_64.llvm.san: hidden symbol `_hb_graphite2_shaper_font_data_destroy' isn't defined
/usr/bin/ld: final link failed: bad value

```

### Steps to reproduce

Above

### Minimal reproduction project

_No response_

--- a/modules/text_server_adv/SCsub
+++ b/modules/text_server_adv/SCsub
@@ -131,9 +131,14 @@ if env["builtin_harfbuzz"]:
         env_harfbuzz.Append(
             CCFLAGS=[
                 "-DHAVE_FREETYPE",
-                "-DHAVE_GRAPHITE2",
             ]
         )
+        if env["graphite"]:
+            env_harfbuzz.Append(
+                CCFLAGS=[
+                    "-DHAVE_GRAPHITE2",
+                ]
+            )
         if env["builtin_freetype"]:
             env_harfbuzz.Prepend(CPPPATH=["#thirdparty/freetype/include"])
         if env["builtin_graphite"] and env["graphite"]:

